### DISCUSSION TITLE
How to set LocalStorage to specific domain

### DISCUSSION THREAD

**Discussion by sagarrajput** [cypress-io/cypress#17696] [General]
I want to visit two different domains(URLs) into 2 different tests, However, I want to set local storage for both of these, If I use **cy.setLocalStorage(key, value)** then localstorage is set only for 1st domain.

**Comment by khvirtru** [comment]
I have the similar question. as below:
![image](https://user-images.githubusercontent.com/25446794/129090534-bb4648d9-9813-43ca-96d5-83b30066f08b.png)


My test scenario is to bypass the Virtru Activation through UI. It is served from a different domain rather than outlook.office.com. My understanding is the cy.window() is for the site that cypress first visited(e.g. in this case, outlook.office.com). And there is no way we can set localStorage for another domain (e.g. in this case, virtru.com).

I cannot switch to the window of the iFrame and setLocalStorage for that window, like below:
```
    getIframeBody().window().then((win)=>{
       // incorrect, the win belongs to outlook.office.com, not the Virtru iFrame as show in the above screenshot
      cy.fixture("localstorage.json")
        .then((data) => {
          for (const key in data){
            cy.log(`${key} ${data[key]}`);
            win.localStorage.setItem(key, data[key])
          }
        });
    })
```
